### ISSUE TITLE
Cannot get GET request to list contents of table

### ISSUE THREAD

**Issue by dpskipper** [mevdschee/php-crud-api#247]
I cannot get this to work, I must be doing something wrong. I've configured and installed this correctly. Doing a HTTP GET on mywebsite.com/api.php returns the correct block of JSON. However doing mywebsite.com/api.php/usrs (name of MySQL db table) returns: `{"Usrs":{}}` There is one column in the table, but this does not show up here. It does appear in the /api.php call, so database must work.

Can you help?

**Comment by mevdschee** [comment]
Thank you for creating this issue. I cannot help you without some more information about what you are doing. For instance, can you be specific about the url you are requesting? You say it is "/api.php/usrs" and that it returns "{"Usrs":{}}", but that seems very unlikely to me. Also can you tell me whether or not there are records in the error log related to this script?

**Comment by dpskipper** [comment]
Hi, as unlikely as it is, that is the full URL that I am using. www.mywebsite.com/api.php/Usrs returns that exact snippet. I will post below the results of www.mywebsite.com/api.php
`{"swagger":"2.0","info":{"title":"MyDB","description":"API generated with [PHP-CRUD-API](https://github.com/mevdschee/php-crud-api)","version":"1.0.0"},"host":"mywebsite.com","basePath":"/api.php","schemes":["http"],"consumes":["application/json"],"produces":["application/json"],"tags":[{"name":"Usrs","description":"List of emails of people."}],"paths":{"/Usrs":{"get":{"tags":["Usrs"],"summary":"List","parameters":[{"name":"exclude","in":"query","description":"One or more related entities (comma separated).","required":false,"type":"string"},{"name":"include","in":"query","description":"One or more related entities (comma separated).","required":false,"type":"string"},{"name":"order","in":"query","description":"Column you want to sort on and the sort direction (comma separated). Example: id,desc","required":false,"type":"string"},{"name":"page","in":"query","description":"Page number and page size (comma separated). NB: You cannot use \"page\" without \"order\"! Example: 1,10","required":false,"type":"string"},{"name":"transform","in":"query","description":"Transform the records to object format. NB: This can also be done client-side in JavaScript!","required":false,"type":"boolean"},{"name":"columns","in":"query","description":"The table columns you want to retrieve (comma separated). Example: posts.*,categories.name","required":false,"type":"string"},{"name":"filter[]","in":"query","description":"Filters to be applied. Each filter consists of a column, an operator and a value (comma separated). Example: id,eq,1","required":false,"type":"array","collectionFormat":"multi","items":{"type":"string"}},{"name":"satisfy","in":"query","description":"Should all filters match (default)? Or any?","required":false,"type":"string","enum":["any"]}],"responses":{"200":{"description":"An array of Usrs","schema":{"type":"array","items":{"type": "object","properties": {"test@test.com": {"type": "string","x-dbtype": "text","x-nullable": false,"maxLength": 65535}}}}}}},"post":{"tags":["Usrs"],"summary":"Create","parameters":[{"name":"item","in":"body","description":"Item to create.","required":true,"schema":{"type": "object","required":["test@test.com"],"properties": {"test@test.com": {"type": "string","x-dbtype": "text","x-nullable": false,"maxLength": 65535}}}}],"responses":{"200":{"description":"Identifier of created item.","schema":{"type":"integer"}}}}},"/Usrs/{id}":{"get":{"tags":["Usrs"],"summary":"Read","parameters":[{"name":"id","in":"path","description":"Identifier for item.","required":true,"type":"string"}],"responses":{"200":{"description":"The requested item.","schema":{"type": "object","properties": {"test@test.com": {"type": "string","x-dbtype": "text","x-nullable": false,"maxLength": 65535}}}}}},"put":{"tags":["Usrs"],"summary":"Update","parameters":[{"name":"id","in":"path","description":"Identifier for item.","required":true,"type":"string"},{"name":"item","in":"body","description":"Properties of item to update.","required":true,"schema":{"type": "object","required":["test@test.com"],"properties": {"test@test.com": {"type": "string","x-dbtype": "text","x-nullable": false,"maxLength": 65535}}}}],"responses":{"200":{"description":"Number of affected rows.","schema":{"type":"integer"}}}},"delete":{"tags":["Usrs"],"summary":"Delete","parameters":[{"name":"id","in":"path","description":"Identifier for item.","required":true,"type":"string"}],"responses":{"200":{"description":"Number of affected rows.","schema":{"type":"integer"}}}},"patch":{"tags":["Usrs"],"summary":"Increment","parameters":[{"name":"id","in":"path","description":"Identifier for item.","required":true,"type":"string"},{"name":"item","in":"body","description":"Properties of item to update.","required":true,"schema":{"type": "object","required":["test@test.com"],"properties": {"test@test.com": {"type": "string","x-dbtype": "text","x-nullable": false,"maxLength": 65535}}}}],"responses":{"200":{"description":"Number of affected rows.","schema":{"type":"integer"}}}}}}}`

**Comment by mevdschee** [comment]
Hi, I checked with a table created with:

    CREATE TABLE `Usrs` (
      `id` int(11) NOT NULL AUTO_INCREMENT,
      PRIMARY KEY (`id`)
    ) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4;

And the URL:

    http://localhost/pi.php/Usrs

The result is:

    {"Usrs":{"columns":["id"],"records":[]}}

For an empty table.

**Comment by dpskipper** [comment]
I had suspected it was something with the database. I must be doing something wrong there. Below are the specs:
```Server: Localhost via UNIX socket
Server type: MariaDB
Server version: 10.0.29-MariaDB-cll-lve - MariaDB Server
Protocol version: 10
Server charset: UTF-8 Unicode (utf8)
Web server

cpsrvd 11.60.0.39
Database client version: libmysql - 5.1.73
PHP extension: mysqli Documentation```

**Comment by dpskipper** [comment]
Here are some more specs:
`
Apache Version	2.2.32
PHP Version	5.4.45
MySQL Version	10.0.29-MariaDB-cll-lve
Architecture	x86_64
Operating System	linux
`

**Comment by dpskipper** [comment]
Ok, it appears i was right all along. Somthing was wrong in my database. I had it wrong all along. Thanks for your help so far.

**Comment by mevdschee** [comment]
>Something was wrong in my database

I'm glad that you have found a solution. Would you mind sharing what was wrong? It may be helpful for anyone else that may run into a similar issue. People tend to search other people's issues to find resolutions for problems.

Thank you for your contribution.

**Comment by dpskipper** [comment]
Sure, it seems I had columns and rows mixed up, a pretty novice mistake :) I had a bunch of columns all named with an email which must of stuffed up something. Once I realised  my mistake, I made one column named it 'email' and then made 5 rows each containing a single email. That fixed it. This might help someone just starting out with an SQL database, who has not figured out what tables ,columns and rows are.
